core/cache/controller.go

```diff
@@ -42,8 +42,9 @@ type Controller struct {
 	// from a type-agnostic viewpoint.
 	manager *residentManager
 
-	config ControllerConfig
-	models map[string]*Model
+	changes <-chan interface{}
```

Comment:
I separated these out so that the channel passed is `ControllerConfig` can be a channel, the `Controller` channel can be declared read-only and the `residentManager` channel can be declared write-only.